### PR TITLE
externaldns: Add ExternalDNS service account logic

### PR DESCRIPTION
**Pull request by sgreene570**
**Manifests: Fix operator Role name**
Makefile:

Use `external-dns-operator` instead of `manager-role` for
the name of the operator's role.

config/rbac/role.yaml:

Run `make manifests` to sync operator role.

cmd/externaldns-operator/main.go -> cmd/external-dns-operator/main.go:
Rename directory for `main.go`.

**externaldns: Add ExternalDNS service account logic**

pkg/operator/controller/externaldns/service_account.go:

New file.
Implements `ensureExternalDNSServiceAccount` and its dependencies,
`currentExternalDNSServiceAccount`, `desiredExternalDNSServiceAccount`,
and `createExternalDNSServiceAccount`.

pkg/operator/controller/externaldns/controller.go:

Get current ExternalDNS resource when reconciles are triggered.

Call `ensureExternalDNSServiceAccount`.

pkg/operator/controller/names.go:

New file for keeping track of operand resource names.
Implements `ExternalDNSServiceAccountName` for now.

pkg/operator/operator.go:
Add RBAC marker for service accounts.

config/rbac/role.yaml:
Regenerate.

### CODE REVIEW

**Comment by Miciah** [comment]
Let's merge this and iterate.
/lgtm

File: Makefile [hunk 1]
```diff
@@ -45,7 +45,7 @@ help: ## Display this help.
 ##@ Development
 
 manifests: ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
-	$(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+	$(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=externaldns-operator webhook paths="./..." output:crd:artifacts:config=config/crd/bases
```
Comment (alebedev87) [inline Makefile:48]: We decided to go for `external-dns-operator` right?

Comment (sgreene570) [reply Makefile]: fixed. Thanks!

**Comment by sgreene570** [comment]
PR should be good to go now

**Pull request merged**